### DISCUSSION TITLE
Running Stig Manager on disconnected environment

### DISCUSSION THREAD

**Discussion by st3v1s** [NUWCDIVNPT/stig-manager#981] [Q&A]
We are trying to start up Stig Manager on a disconnected classified network.  We exported our NIPR containers and imported into the disconnected docker environment.  We changed the IP Addresses in the docker-compose.yml.  Also set the STIGMAN_INIT_IMPORT_STIGS=false.  

I was looking at the documentation and noticed the STIGMAN_CLIENT_EXTRA_SCOPES talks about offline access.  Can you explain how that is used?  Also are there any other vaiables I should set for the type of environment we are trying to run in?

Thanks for your time!

**Comment by cd-rite** [comment]
HI @st3v1s The offline_access scope is part of the OIDC specification, and just means the app would like to use refresh tokens. Not all OIDC providers require it to be specified, which is why that envvar allows you to add it. (I don't think Keycloak requires it, if it has refresh tokens enabled). 
As long as everything is pointed to the right places in the new environment, things should work ok...
You might need to change what you provide with NODE_EXTRA_CA_CERTS if you have different cert chains in the new environment.

**Reply by st3v1s** [reply]
We are seeing errors with the api container trying to connect to the 50001 port...if I change to 3306 it shows same ECONNREACHED  success:  false.

I am able to load the Welcome to KeyCloak page in the browser by going to http://IP:8080  but when I select Admin Console, it fails and says HTTPS required.  Is this due to the admin user and password not created during startup?

Any idea what I could I look into for these errors?

**Comment by cd-rite** [comment]
HI @st3v1s 3306 would probably be the port that the API connects to the DB on, and not what you would try to hit STIGMan on. 
Keycloak can be configured many different ways depending on your requirements, and is not part of our project, so we can't provide too much help there. There were some changes last year to the Keycloak project that might need to be accounted for, I'd check their docs for help there though.  We do have a sample configuration you might be able to adapt to your purposes: https://github.com/NUWCDIVNPT/stigman-orchestration

**Comment by st3v1s** [comment]
Still banging my head on the stig manager api not starting.  I was wondering do I need to export the volume folder called stigdata as I don't understand how this will work on a disconnected system.  Specifically we are trying to stand up on SIPR.  Is there a github resource for SIPR?
I appreciate any help you can provide.
